### PR TITLE
feat(libero): add an opt-in data flywheel

### PR DESCRIPTION
**Pull request by huanming99**
## Summary

This PR adds a minimal, opt-in LIBERO data flywheel to RPent:

- record action-aligned policy observations, actions, rewards, termination flags,
  Primitive ownership, and VLA proposals during normal evaluation;
- keep finalized raw episodes immutable and retain both successful and failed runs;
- validate raw episodes and export successful training prefixes to LeRobot;
- launch native Pi0.5 SFT through an explicit official RLinf checkout.

Collection is disabled by default. The existing fast VLA chunk path remains
unchanged when Flywheel collection is not enabled.

## Usage

Enable collection on a normal LIBERO evaluation:

```bash
python -m rpent.cli.main \
  --robot libero \
  --suite libero_object_swap \
  --task 2 \
  --seed 1 \
  --collect-flywheel-data \
  --flywheel-root datacollection \
  ...
```

Validate and export successful episodes:

```bash
rpent-flywheel validate datacollection/raw/libero/<suite>/task_<id>/seed_<seed>/<episode>

rpent-flywheel export-lerobot \
  --data-root datacollection \
  --suite libero_object_swap \
  --task 2 \
  --output-root datasets
```

Run Pi0.5 SFT with official RLinf:

```bash
rpent-flywheel train-rlinf \
  --dataset datasets/<dataset-id> \
  --checkpoint /path/to/pi05 \
  --rlinf-root /path/to/RLinf \
  --output-dir /path/to/new-training-output \
  --max-steps 1000 \
  --save-interval 100 \
  --cuda-device 0
```

## Data contract

- An episode is first written to a `.partial` directory and atomically
  published only after validation.
- Each action is aligned with its policy input observation and resulting
  observation.
- Scripted actions retain their Primitive ID; VLA actions additionally retain
  the VLA chunk ID and proposal index.
- Environment termination is the success label. Failed episodes remain in raw
  storage, while the current SFT exporter selects successful training prefixes.
- Export and training create derived outputs and do not overwrite raw episodes
  or the input checkpoint.

## Validation

- Full test suite: `13 passed, 1 skipped`.
- Ruff, Python compilation, `git diff --check`, Hydra config composition, clean
  wheel build, and installed-wheel CLI checks passed.
- Current collection runs finalized five action-aligned episodes with no stale
  `.partial` directories, including VLA and scripted Primitive execution.
- A known successful 138-action LIBERO trajectory was replayed at the same
  deterministic reset state through the current environment and writer. The
  environment again returned `terminated=True` at step 138; the resulting raw
  episode contained five action-producing Primitive calls and ten VLA proposals
  and passed installed-wheel validation.
- The installed-wheel exporter produced and reopened a one-episode, 138-frame
  LeRobot dataset. Raw-file SHA-256 manifests were identical before and after
  export.
- Official `RLinf@dd92c62857da4c67aa5e7c36f731c0d6a121f6d7` consumed that dataset,
  completed exactly one Pi0.5 optimizer step, and saved a readable complete
  checkpoint. Loss (`0.00112`) and gradient norm (`0.433`) were finite; sampled
  vision, language, action-expert, and projection tensors changed, while the
  input checkpoint checksum remained unchanged.

This validation establishes the data and training integration, not model-quality
improvement or a success-rate claim.

### CODE REVIEW

**Comment by huanming99** [comment]
Closing this initial draft because GitHub did not synchronize its cached head after the branch was updated with the required Ruff formatting fix. Recreating from the current branch head.
